hazelcast/src/main/java/com/hazelcast/internal/util/XmlUtil.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util;
+
+import static com.hazelcast.internal.nio.IOUtil.closeResource;
+
+import java.io.StringReader;
+import java.io.StringWriter;
+
+import javax.annotation.Nullable;
+import javax.xml.XMLConstants;
+import javax.xml.transform.ErrorListener;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.SchemaFactory;
+
+import org.xml.sax.SAXException;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+/**
+ * Utility class for XML processing.
+ */
+public final class XmlUtil {
+
+    /**
+     * System property name which allows ignoring failures during enabling the XML External Entity protection.
+     * This property should only be used as a last
+     * resort. Hazelcast uses the XXE protection by setting properties {@link XMLConstants#ACCESS_EXTERNAL_DTD} and
+     * {@link XMLConstants#ACCESS_EXTERNAL_SCHEMA}. These properties are supported in modern XML processors (JAXP 1.5+, Java
+     * 8+). Old JAXP implementations on the classpath (e.g. Xerces, Xalan) may miss the support and they throw exception
+     * during enabling the XXE protection. Setting this system property to true suppresses/ignores such Exceptions.
+     */
+    public static final String SYSTEM_PROPERTY_IGNORE_XXE_PROTECTION_FAILURES = "com.hazelcast.ignoreXxeProtectionFailures";
```

Comment:
nit: all our cluster properties skip the `com.` part in their name (eg. `hazelcast.partition.count`), consider removing it?

Reply:
good point, will remove it